src/components/chip-group/chip-group.tsx

```diff
@@ -0,0 +1,245 @@
+    updateHostInteraction(this);
+  }
+
+  componentDidLoad(): void {
+    if (!this.sorted) {
+      this.items = this.sortItems(this.items);
+      this.sorted = true;
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("calciteInternalChipKeyEvent")
+  calciteInternalChipKeyEvent(event: CustomEvent): void {
+    const item = event.detail.item;
+    const parent = event.detail.parent as HTMLCalciteChipGroupElement;
+    if (this.el === parent) {
+      const { key } = item;
+      const itemToFocus = event.target;
+      const isFirstItem = this.itemIndex(itemToFocus) === 0;
+      const isLastItem = this.itemIndex(itemToFocus) - 1 === this.items.length - 1;
+      switch (key) {
+        case "ArrowRight":
+          if (isLastItem) {
+            this.focusFirstItem();
+          } else {
+            this.focusNextItem(itemToFocus);
+          }
+          break;
+        case "ArrowLeft":
+          if (isFirstItem) {
+            this.focusLastItem();
+          } else {
+            this.focusPrevItem(itemToFocus);
+          }
+          break;
+        case "Home":
+          this.focusFirstItem();
+          break;
+        case "End":
+          this.focusLastItem();
+          break;
+      }
+    }
+    event.stopPropagation();
+  }
+
+  @Listen("calciteChipClose")
+  calciteChipListener(event: CustomEvent): void {
+    const item = event.target;
+    let updatedItems = [];
+    if (this.items.includes(item)) {
+      updatedItems = this.items.filter((el) => el !== item);
+      if (this.itemIndex(item) > 0) {
+        this.focusPrevItem(item);
+      } else if (this.itemIndex(item) === 0) {
+        this.focusNextItem(item);
+      } else {
+        this.focusFirstItem();
+      }
+    }
+    this.items = updatedItems;
+  }
+
+  @Listen("calciteInternalChipRegister")
+  registerCalciteChip(event: CustomEvent): void {
+    const item = {
+      item: event.target as HTMLCalciteChipElement,
+      parent: event.detail.parent as HTMLCalciteChipGroupElement,
+      position: (event.detail.position as number) || 0
+    };
+    if (!this.items.includes(item) && item.parent && this.el === item.parent) {
+      this.items.push(item);
+    }
+    event.stopPropagation();
+  }
+
+  @Listen("calciteInternalChipSelect")
+  updateActiveItemOnChange(event: CustomEvent): void {
+    this.calciteChipInternalSelectionChange.emit({
+      requestedChip: event.detail.requestedChip
+    });
+    const selectedItems = Array.from(this.el.getElementsByTagName("calcite-chip")).filter(
+      (el) => el.selected
+    );
+    this.calciteChipGroupChange.emit({
+      selectedItems
+    });
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
```

Comment:
All these private methods look great but potentially really useful to several components. What about pulling them out into utilities?

Reply:
💯 agreed - Accordion, Dropdown, Stepper use exact or very similar versions of these as well so making a shared utility makes sense.

Reply:
💯 PR opened - https://github.com/Esri/calcite-components/pull/6086